### PR TITLE
Add lazy initialization to AdcController

### PR DESCRIPTION
**Pull request by josesimoes**
## Description
- Add lazy initialization to I2cController.
- Also changed controllersCollection making it accessible as property.
- Add AdcControllerManager allowing multiple ADC controllers.

## Motivation and Context
- Make sure that the variables do exist when being used. Using the previous code was causing issues because, on occasions, the related types could still be unresolved.
- Decorate fields with DebuggerBrowsableState to hide them on debug session.

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

Signed-off-by: José Simões <jose.simoes@eclo.solutions>

### CODE REVIEW

**Comment by josesimoes** [comment]
@steffalk please hold the review.

**Comment by josesimoes** [comment]
@steffalk it's OK to review this now.

File: source/Windows.Devices.Adc/AdcController.cs [hunk 1]
```diff
@@ -14,27 +14,66 @@ namespace Windows.Devices.Adc
...
+            {
+                s_deviceCollection = value;
+            }
+        }
 
         internal AdcController(string adcController)
         {
             // check if this device is already opened
-            if (!s_deviceCollection.Contains(adcController))
+            if (!AdcControllerManager.ControllersCollection.Contains(adcController))
```
Comment (steffalk) [inline source/Windows.Devices.Adc/AdcController.cs]: Is it sure there can't be issues due to different casing of parameter adcController and the values stored in the HashSet?

Comment (josesimoes) [reply source/Windows.Devices.Adc/AdcController.cs]: good question...
maybe we should enforce this to upper case... what do you think?

File: source/Windows.Devices.Adc/AdcController.cs [hunk 2]
```diff
@@ -121,18 +160,21 @@ internal AdcController(string adcController)
...
-            string controllers = GetDeviceSelector();
-            string[] devices = controllers.Split(',');
+            string controllersAqs = GetDeviceSelector();
+            string[] controllers = controllersAqs.Split(',');
 
-            if(devices.Length > 0)
+            if(controllers.Length > 0)
             {
-                if (s_deviceCollection.Contains(devices[0]))
+                if (AdcControllerManager.ControllersCollection.Contains(controllers[0]))
```
[the remark this answers is not in the document]
Comment (josesimoes) [reply source/Windows.Devices.Adc/AdcController.cs]: For ADC there is only one controller yes. 
But (like with the other class libs that using controllers) the default is (by design) the first controller in the list of existing controllers so the 0 index is correct there.

File: source/Windows.Devices.Adc/AdcController.cs (hunk 1 above)
Comment (josesimoes) [reply source/Windows.Devices.Adc/AdcController.cs]: In this particular case (ADC) you can't specify an ADC controller with a string. See [here](https://docs.microsoft.com/en-us/uwp/api/windows.devices.adc.adccontroller.getcontrollersasync).

So we are good as it is now, the string is internal and coming from the native code. Definitely hardcoded!

**Pull request merged**